extensions/amp-date-countdown/1.0/date-countdown.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {useEffect, useState} from '../../../src/preact';
+import {useResourcesNotify} from '../../../src/preact/utils';
+import {useAmpContext} from '../../../src/preact/context';
+
+const NAME = 'DateCountdown';
+
+const DEFAULT_OFFSET_SECONDS = 0;
+const DEFAULT_LOCALE = 'en';
+const DEFAULT_WHEN_ENDED = 'stop';
+const DEFAULT_BIGGEST_UNIT = 'DAYS';
+const DELAY = 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_DAY = 24 * 60 * 60 * 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_HOUR = 60 * 60 * 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_MINUTE = 60 * 1000;
+
+/** @const {number} */
+const MILLISECONDS_IN_SECOND = 1000;
+
+const LOCALE_WORD = {
+  'de': ['Jahren', 'Monaten', 'Tagen', 'Stunden', 'Minuten', 'Sekunden'],
+  'en': ['Years', 'Months', 'Days', 'Hours', 'Minutes', 'Seconds'],
+  'es': ['años', 'meses', 'días', 'horas', 'minutos', 'segundos'],
+  'fr': ['ans', 'mois', 'jours', 'heures', 'minutes', 'secondes'],
+  'id': ['tahun', 'bulan', 'hari', 'jam', 'menit', 'detik'],
+  'it': ['anni', 'mesi', 'giorni', 'ore', 'minuti', 'secondi'],
+  'ja': ['年', 'ヶ月', '日', '時間', '分', '秒'],
+  'ko': ['년', '달', '일', '시간', '분', '초'],
+  'nl': ['jaar', 'maanden', 'dagen', 'uur', 'minuten', 'seconden'],
+  'pt': ['anos', 'meses', 'dias', 'horas', 'minutos', 'segundos'],
+  'ru': ['год', 'месяц', 'день', 'час', 'минута', 'секунда'],
+  'th': ['ปี', 'เดือน', 'วัน', 'ชั่วโมง', 'นาที', 'วินาที'],
+  'tr': ['yıl', 'ay', 'gün', 'saat', 'dakika', 'saniye'],
+  'vi': ['năm', 'tháng', 'ngày', 'giờ', 'phút', 'giây'],
+  'zh-cn': ['年', '月', '天', '小时', '分钟', '秒'],
+  'zh-tw': ['年', '月', '天', '小時', '分鐘', '秒'],
+};
+
+/**
+ * @param {!DateCountdownPropsDef} props
+ * @return {PreactDef.Renderable}
+ */
+export function DateCountdown({
+  endDate,
+  timeleftMs,
+  timestampMs,
+  timestampSeconds,
+  offsetSeconds = DEFAULT_OFFSET_SECONDS,
+  whenEnded = DEFAULT_WHEN_ENDED,
+  locale = DEFAULT_LOCALE,
+  biggestUnit = DEFAULT_BIGGEST_UNIT,
+  render,
+  children,
+}) {
+  useResourcesNotify();
+  const {playable} = useAmpContext();
```

Comment:
For now it has to be `const {'playable': playable} = useAmpContext()`.

Reply:
Won't be needed after https://github.com/ampproject/amphtml/pull/30102

Reply:
Yep, sounds good!  Updated for now.